tests/integration/test_charm.py

```diff
@@ -261,3 +262,65 @@ async def test_log_level_change(ops_test: OpsTest):
     await ops_test.model.wait_for_idle(
         apps=[APP_NAME], status="active", timeout=1000, idle_period=30
     )
+
+
+@pytest.mark.group(1)
+@pytest.mark.abort_on_fail
+async def test_dashboard_status_changes(ops_test: OpsTest):
+    """Test HTTPS access to each dashboard unit."""
+
+    logger.info("Breaking opensearch connection")
+    await ops_test.juju("remove-relation", "opensearch", "opensearch-dashboards")
+    await ops_test.model.wait_for_idle(apps=[OPENSEARCH_APP_NAME], status="active", timeout=1000)
+
+    async with ops_test.fast_forward("30s"):
+        await ops_test.model.wait_for_idle(apps=[APP_NAME], status="blocked")
+
+    assert await check_full_status(
+        ops_test, status="blocked", status_msg="Opensearch connection is missing"
+    )
+
+    logger.info("Checking if Dashboards have become unavailable")
+    assert all_dashboards_unavailable(ops_test, https=True)
+
+    logger.info("Restoring Opensearch connection")
+    await ops_test.model.integrate(APP_NAME, OPENSEARCH_APP_NAME)
+    await ops_test.model.wait_for_idle(
+        apps=[APP_NAME, OPENSEARCH_APP_NAME], status="active", timeout=1000
+    )
+    assert ops_test.model.applications[APP_NAME].status == "active"
+    assert all(
+        unit.workload_status == "active" for unit in ops_test.model.applications[APP_NAME].units
+    )
+
+    logger.info("Checking if Dashboards is available again")
+    opensearch_relation = get_relations(ops_test, OPENSEARCH_RELATION_NAME)[0]
+    assert access_all_dashboards(ops_test, opensearch_relation, https=True)
```

Comment:
1. I believe we should pass the `opensearch_relation`'s ID not the object 
2. 
```suggestion
    assert await access_all_dashboards(ops_test, opensearch_relation, https=True)
```

Reply:
You are totally right, thanks for noticing!!!!